src/test/groovy/org/prebid/server/functional/tests/privacy/GdprAuctionSpec.groovy

```diff
@@ -0,0 +1,207 @@
+        def bidRequest = getGdprBidRequest(validConsentString)
+
+        and: "Save account config into DB"
+        def privacy = new AccountPrivacyConfig(gdpr: gdprConfig)
+        def accountConfig = new AccountConfig(privacy: privacy)
+        def account = new Account(uuid: bidRequest.site.publisher.id, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        privacyPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain masked values"
+        def bidderRequests = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequests.device?.geo == maskGeo(bidRequest)
+
+        where:
+        gdprConfig << [new AccountGdprConfig(enabled: true),
+                       new AccountGdprConfig(enabled: false, channelEnabled: [(ChannelType.WEB): true])]
+    }
+
+    @Unroll
+    def "PBS should not apply gdpr when privacy.gdpr.channel-enabled.app or privacy.gdpr.enabled = false in account config"() {
+        given: "Default basic generic BidRequest"
+        def validConsentString = new TcfConsent.Builder()
+                .setPurposesLITransparency(BASIC_ADS)
+                .addVendorLegitimateInterest([GENERIC_VENDOR_ID])
+                .build()
+        def bidRequest = getGdprBidRequest(DistributionChannel.APP, validConsentString)
+
+        and: "Save account config into DB"
+        def privacy = new AccountPrivacyConfig(gdpr: gdprConfig)
+        def accountConfig = new AccountConfig(privacy: privacy)
+        def account = new Account(uuid: bidRequest.app.publisher.id, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain not masked values"
+        def bidderRequests = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequests.device?.geo?.lat == bidRequest.device.geo.lat
+        assert bidderRequests.device?.geo?.lon == bidRequest.device.geo.lon
+
+        where:
+        gdprConfig << [new AccountGdprConfig(enabled: true, channelEnabled: [(ChannelType.APP): false]),
+                       new AccountGdprConfig(enabled: false)]
+    }
+
+    @Unroll
+    def "PBS should not apply gdpr when privacy.gdpr.channel-enabled.web or privacy.gdpr.enabled = false in account config"() {
+        given: "Default basic generic BidRequest"
+        def validConsentString = new TcfConsent.Builder()
+                .setPurposesLITransparency(BASIC_ADS)
+                .addVendorLegitimateInterest([GENERIC_VENDOR_ID])
+                .build()
+        def bidRequest = getGdprBidRequest(validConsentString)
+
+        and: "Save account config into DB"
+        def privacy = new AccountPrivacyConfig(gdpr: gdprConfig)
+        def accountConfig = new AccountConfig(privacy: privacy)
+        def account = new Account(uuid: bidRequest.site.publisher.id, config: accountConfig)
```

Comment:
maybe it would be better to create private method in that spec to construct and return that object? As you are using the same 3 code rows among 4 tests in that spec.